broadinstitute/scp-ingest-pipeline
# Update ingest to extract processed gene expression as mtx (SCP-5127)

This is a "happy path" implementation of extracting processed gene expression from h5ad.

Changes impacting Rails integration:
- Extracted files will be delocalized to a "subdirectory" that is now a concatenation of the h5ad file's `{study_accession}_{study_file_id}`.
- All extracted files are now gzipped.

To test:
Run anndata ingest using an example file from our development reference data GCP bucket.

Before you start, check the study bucket for delocalized output
`https://console.cloud.google.com/storage/browser/fc-2f8ef4c0-b7eb-44b1-96fe-a07f0ea9a982/`
If there is an "_scp_internal" directory, delete it before you start.

Activate the scp-ingest-pipeline repo virtualenv
From the ingest directory of the scp-ingest-pipeline repo, perform this setup:
```
source ../scripts/setup_mongo_dev.sh <path to your Github token file>
unset BARD_HOST_URL
export BYPASS_MONGO_WRITES='yes'
```

If you get an authentication error from gcloud, run the following:
`gcloud auth application-default login --no-launch-browser`

Run the anndata ingest job from the ingest directory of the scp-ingest-pipeline repo:
`python ingest_pipeline.py  --study-id 5d276a50421aa9117c982845 --study-file-id 5dd5ae25421aa910a723a337 ingest_anndata --ingest-anndata --anndata-file gs://fc-2f8ef4c0-b7eb-44b1-96fe-a07f0ea9a982/test_Data/ingest_manual_test/anndata_test.h5ad --extract "['cluster', 'metadata', 'processed_expression']" --obsm-keys "['X_tsne']"`

Confirm that locally, in the ingest directory, you find intermediate ingest files:
> h5ad_frag.barcodes.processed.tsv.gz
> h5ad_frag.cluster.X_tsne.tsv.gz
> h5ad_frag.features.processed.tsv.gz
> h5ad_frag.matrix.processed.mtx.gz
> h5ad_frag.metadata.tsv.gz

Confirm that the job ran successfully by checking that the above outputs delocalized to (note gsURL contains dummy accession "SCPdev"):
`https://console.cloud.google.com/storage/browser/fc-2f8ef4c0-b7eb-44b1-96fe-a07f0ea9a982/_scp_internal/anndata_ingest/SCPdev_5dd5ae25421aa910a723a337`
